src/grammar.coffee

```diff
@@ -629,16 +629,16 @@ grammar =
   # The variants of *try/catch/finally* exception handling blocks.
   Try: [
     o 'TRY Block',                              -> new Try $2
-    o 'TRY Block Catch',                        -> new Try $2, $3[0], $3[1]
-    o 'TRY Block FINALLY Block',                -> new Try $2, null, null, $4
-    o 'TRY Block Catch FINALLY Block',          -> new Try $2, $3[0], $3[1], $5
+    o 'TRY Block Catch',                        -> new Try $2, $3
+    o 'TRY Block FINALLY Block',                -> new Try $2, null, $4, LOC(3)(new Literal $3)
+    o 'TRY Block Catch FINALLY Block',          -> new Try $2, $3, $5, LOC(4)(new Literal $4)
   ]
 
   # A catch clause names its error and runs a block of code.
   Catch: [
-    o 'CATCH Identifier Block',                 -> [$2, $3]
-    o 'CATCH Object Block',                     -> [LOC(2)(new Value($2)), $3]
-    o 'CATCH Block',                            -> [null, $2]
+    o 'CATCH Identifier Block',                 -> new Catch $3, $2
```

Comment:
By introducing a new `Catch` node class, we are able to use the standard AST methods as well as not have to do any additional work to preserve the location data of the `catch` keyword